code/src/OSFramework/OSStructure/GridChanges.ts

```diff
@@ -0,0 +1,8 @@
+/// <reference path="../../OSFramework/OSStructure/changesDone.ts" />
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+namespace OSFramework.OSStructure {
+    export class GridChanges extends OSFramework.OSStructure.ChangesDone {
+        hasInvalidLines: boolean;
+        invalidLinesJSON: string;
```

Comment:
TOP!

Reply:
Why create a second class? and not add these fields to the class ChangesDone?

Reply:
this is a feature data... invalidLines comes from ValidationFeature...

DataSource treats only changes, deletion and insertion....

This second class can be incremented without affecting DataSource